site/zenodo_rdm/migrator/transform/records.py

```diff
@@ -0,0 +1,206 @@
+            r["embargo"] = {
+                "until": record["embargo_date"],
+                "active": True,
+            }
+        return r
+
+    def _metadata(self, entry):
+        """Transform the metadata of a record."""
+
+        def _creators(data):
+            ret = []
+            for c in data:
+                r = {"person_or_org": {"type": "personal"}}
+                if c.get("affiliation"):
+                    r["affiliations"] = [{"name": c["affiliation"]}]
+                if c.get("orcid"):
+                    r["person_or_org"]["identifiers"] = [
+                        {"scheme": "orcid", "identifier": c["orcid"]},
+                    ]
+                name = HumanName(c["name"])
+                r["person_or_org"]["given_name"] = name.first
+                r["person_or_org"]["family_name"] = name.surnames
+                # autocompleted by RDM Metadata schema
+                r["person_or_org"]["name"] = f"{name.surnames}, {name.first}"
+
+                ret.append(r)
+            return ret
+
+        def _resource_type(data):
+            t = data["type"]
+            st = data.get("subtype")
+            return {"id": f"{t}-{st}"} if st else {"id": t}
+
+        record = entry["json"]
+        r = {
+            "title": record["title"],
+            "description": record["description"],
+            "publication_date": record["publication_date"],
+            "resource_type": _resource_type(record["resource_type"]),
+            "creators": _creators(record["creators"]),
+        }
+
+        return r
+
+    def _custom_fields(self, entry):
+        """Transform custom fields."""
+        metadata = entry.get("json", {})
+        cf = {
+            "journal:journal": {
+                "title": metadata.get("journal", {}).get("title"),
+                "issue": metadata.get("journal", {}).get("issue"),
+                "pages": metadata.get("journal", {}).get("pages"),
+                "volume": metadata.get("journal", {}).get("volume"),
+                "issn": metadata.get("journal", {}).get("issn"),
+            },
+            "meeting:meeting": {
+                "acronym": metadata.get("meeting", {}).get("acronym"),
+                "dates": metadata.get("meeting", {}).get("dates"),
+                "place": metadata.get("meeting", {}).get("place"),
+                "session_part": metadata.get("meeting", {}).get("session_part"),
+                "session": metadata.get("meeting", {}).get("session"),
+                "title": metadata.get("meeting", {}).get("title"),
+                "url": metadata.get("meeting", {}).get("url"),
+            },
+            "imprint:imprint": {
+                "isbn": metadata.get("imprint", {}).get("isbn"),
+                "publisher": metadata.get("imprint", {}).get("publisher"),
+                "place": metadata.get("imprint", {}).get("place"),
+                # QUESTION: the following two are not found in Zenodo
```

Comment:
@alejandromumo I'm a bit delayed in catching up with this. Are these fields new, or the match any old Zenodo ones?

Reply:
They match Zenodo's :

```
"part_of.title": "custom_fields.imprint:imprint.title",
"part_of.pages": "custom_fields.imprint:imprint.pages",
```

See this record on [Sandbox](https://sandbox.zenodo.org/api/records/1174530). 
It has all the `book` fields and these two (`title` and `pages`) are set to `part_of`

Reply:
Fixed, Thanks a lot!